### PR TITLE
GitHub Actions, Python 3.9+ Support, Jupyterlab Extensions

### LINKED ISSUE #899: Incorporate cell execution time
**Is your feature request related to a problem? Please describe.**
It would be nice to know how long a certain cell took to run

**Describe the solution you'd like**
Something like this

https://github.com/ipython-contrib/jupyter_contrib_nbextensions/tree/7672d429957aaefe9f2e71b15e3b78ebb9ba96d1/src/jupyter_contrib_nbextensions/nbextensions/execute_time

**Describe alternatives you've considered**
Nothing

**Additional context**

### LINKED ISSUE #954: Unpin `ipywidgets` in the future
*Please refer to our [FAQ](https://ark-analysis.readthedocs.io/en/latest/_rtd/faq.html) and look at our [known issues](https://github.com/angelolab/ark-analysis/issues) before opening a bug report.*

Unpin `ipywidgets` once a new release where `ipywidgets` > 8.0.6 is made, or when a new release of `comm` is made.

### LINKED ISSUE #965: Document supported python versions, expand versions
**Is your feature request related to a problem? Please describe.**
The current `ark` build only supports python 3.8. If you try and pip install using 3.9, this unintuitive error message comes up: 
image

**Describe the solution you'd like**
We should more clearly indicate which versions are supported. We should also begin testing multiple versions of python to make the subsequent process of switching/deprecating relatively seamless.

### PR DESCRIPTION
**Pull request by srivarra**
**If you haven't already, please read through our [contributing guidelines](https://ark-analysis.readthedocs.io/en/latest/_rtd/contributing.html) before opening your PR**

**What is the purpose of this PR?**

- Closes #965.
- Closes #899.
- Closes #954.

Adds a Conda environment using Python 3.11.
The docker environment now uses a slim Python 3.11 Image.

**How did you implement your changes**

## Environment Changes

Added the `environment.yml` file, adjusted the `Dockerfile`, made changes to the `README` and reformatted the `setup.py` file.
Support for Python 3.9 to 3.11.

Added the following extensions:
- [jupyterlab-lsp](https://github.com/jupyter-lsp/jupyterlab-lsp)
- [jupyterlab-git](https://github.com/jupyterlab/jupyterlab-git)
- [jupyterlab-skip-traceback](https://github.com/deshaw/jupyterlab-skip-traceback)
- [lckr-jupyterlab-variableinspector](https://github.com/lckr/jupyterlab-variableInspector)

These extensions are not dependencies of Ark, they are installed alongside Ark in the Conda `environment.yml` and the Docker image Dockerfile, and can be optionally installed with `pip install ark-analysis[lab_ext]` or `pip install .[lab_ext]` if you are in the ark directory.

## Testing Changes
- Added a Python Script which runs as the first job in the CI workflow. This will grab the huggingface-dataset cache, and will verify and update it if necessary. (In the future we can add a drop down menu for us to be able to select a specific commit when working with PRs which require dataset changes). The cache is now cross-os enabled, so 1 cache for all tests. Tests are only able to read the cache, not write to it.
- The fiber segmentation tests do not require the example dataset anymore, so these should be a bit faster. 
- Reduced the cell count for `dimensionality_reduction_test.py` from 300 to 50.

Removed unused imports in `src/ark/`,  `tests/`, and `templates/`

Sorted all imports with `isort`, removed unused imports.

**Remaining issues**
Merge the following before this pr:
- [x] Merge #998
- [x] Merge #997
- [x] Merge #992
- [x] Merge #972 / #986

### CODE REVIEW

**Review by alex-l-kong — APPROVED** [review]
Looks good

**Review by ngreenwald — CHANGES_REQUESTED** [review]
We should get the sequencing right for all of this. If @cliu72  wants the updated version for her paper to to be 3.9+, then we should wait on merging in #985 until after this. 

If not, then we should make a new release once her other requests get merged in, then #985, then this.

**Comment by cliu72** [comment]
> We should get the sequencing right for all of this. If @cliu72 wants the updated version for her paper to to be 3.9+, then we should wait on merging in #985 until after this.
> 
> If not, then we should make a new release once her other requests get merged in, then #985, then this.

Yeah I think it'd be good for the updated version of my paper to be 3.9+, since pip install doesn't work anymore.

File: environment.yml [hunk 1]
```diff
@@ -14,7 +14,13 @@ dependencies:
...
+  - jupyterlab-lsp
+  - python-lsp-server[all]
+  - jupyterlab-git
+  - jupyterlab_execute_time
   - pip:
-    - alpineer==0.1.6
-    - pyFlowSOM==0.1.14
+    - jupyterlab-skip-traceback
+    - lckr-jupyterlab-variableinspector
+    - alpineer==0.1.7
```
Comment (camisowers) [inline environment.yml:24]: Should we bump alpineer to 0.1.8 instead? [https://github.com/angelolab/alpineer/releases/tag/v0.1.8](https://github.com/angelolab/alpineer/releases/tag/v0.1.8)

**Review by camisowers — CHANGES_REQUESTED** [review]
Minor comment about the alpineer version, but other than that it looks great.

**Comment by srivarra** [comment]
@ngreenwald This is good to go. 

@camisowers @alex-l-kong   Added the fix in for `watershed`'s argument `dtype`.

**Review by ngreenwald — APPROVED** [review]
Looks great! Big upgrades

**Review by camisowers — APPROVED** [review]

**Review by camisowers — APPROVED** [review]

**Pull request merged**